### ISSUE TITLE
VideoJS 5 with fluid option is unable to start (undefined function on ima3.js)

### ISSUE THREAD

**Issue by spodlecki** [googleads/videojs-ima#316]
# Bug
The new responsive css for VideoJS causes tap / touch events to never fire. The iFrame seems to be above the video player listeners.

You can "fix" this by setting the z-index of the `video` element to say, 1200, and then click. It'll start the ad, but there will need to be code that re-hides the video. If you keep the z-index at 1200, the ad is ran below an empty video element.

```
Uncaught TypeError: Cannot read property 'apply' of undefined  | # => ima3.js#288
```

# Recreate
* Setup a video with IMA (used Sample app in repo)
* Set video options: `fluid: true`
* View in a mobile device (OS doesn't matter as far as I can tell)
* Tap the video and note that nothing will happen

## Work around

To get around this, I've updated the z-indexes of the items videojs is listening to. I did not test this without a poster, I'm always running a poster image.

```
player.posterImage.el().style.zIndex = 1400
player.bigPlayButton.el().style.zIndex = 1410
player.loadingSpinner.el().style.zIndex = 1420
```

## JS Bin
https://jsbin.com/vofigafuno/edit?css,js,console,output

**Comment by shawnbuso** [comment]
Hi there,

Is your jsbin meant to reproduce the issue? It doesn't look like it includes your workaround, but I loaded it up on my Android device and I am able to tap on the video to play it.

**Comment by spodlecki** [comment]
It is to reproduce the issue. Just tested on Android 5.1.1 (native browser) and it still exists. It is possible that you've managed to touch an uncovered area? Not sure.

Steps that I'm using:
* Visit JS Bin
* Wait for entire page to load
* Touch play icon (or in upper left quadrant of the video)

It should just sit and do nothing.

A screenshot from a Google Dev Console highlighting the iframe being over other items:
![js_bin_-_collaborative_javascript_debugging](https://cloud.githubusercontent.com/assets/1273570/22382995/9efd408c-e48d-11e6-88ec-e72e162618bc.png)

**Comment by shawnbuso** [comment]
Hmm I'm still not able to repro, though I'm on Android 7. I'll see if I have an Android 5 phone here to test with.

**Comment by spodlecki** [comment]
Tested on `iPhone 6S Plus, iOS 10.2` and it works as expected

**Comment by shawnbuso** [comment]
Thanks for the additional data point - still digging around for an Android 5 phone here. I'll try in the simulator to see if that works.

**Comment by shawnbuso** [comment]
Sure thing, feel free to give it a go. We take PRs as long as you can sign our CLA here: https://cla.developers.google.com

**Comment by jsonMartin** [comment]
I'm having this issue too, FYI. 

It is happening to me when testing in responsive iPhone mode in the iPad. The iFrame for the ad is eating the touch event, and click/touch handlers on the parent video container aren't getting called which means the initialize and request ad code is never happening.

@spodlecki , your suggestion of setting the z-index of the big play button is the only thing I've found that resolves this issue for me. If I use it, the play button eats the touch event before the injected ad iFrame gets it - otherwise, I'll get the ima3.js error.

**Comment by shawnbuso** [comment]
Testing with videojs-ima 1.0, it looks like this is resolved. Are you still able to repro with the latest update?

**Comment by ypavlotsky** [comment]
Still fixed in 1.1.0 Closing this out.
